### PR TITLE
Adds a bind mount from host to /usr/share/stormond/ directory on pmon container

### PR DESCRIPTION
**Pull request by assrinivasan**
#### Why I did it
This is a part of a larger commit that adds a storage monitoring daemon to SONiC. This commit is needed to that storagemon daemon can save FS stats JSON file to disk in the event of planned reboots and unintentional powercycles.

[SONiC Storage Monitoring Daemon HLD](https://github.com/sonic-net/SONiC/pull/1481)

Also added `psutil` module to perform process and system monitoring operations pythonically.

##### Work item tracking
- Microsoft ADO **(number only)**: 17468992

#### How I did it
- Added a rule to create the directory on the pmon container
- Added a rule to create a bind-mount

#### How to verify it

Flash a switch with image containing the changes. 
Run:

```
root@str2-7050cx3-acs-13:~# docker inspect pmon
```

and verify the following parameters in the "mounts" section:
```
        "Mounts": [
            .
            .
            .
            {
                "Type": "bind",
                "Source": "/host/pmon/stormond",
                "Destination": "/usr/share/stormond",
                "Mode": "rw",
                "RW": true,
                "Propagation": "rprivate"
            },
            .
            .
            .
        ]
```

Additionally, you can run the following steps:

1. Navigate to /host/pmon/stormond directory on the host and create an empty file
2. Enter the pmon container and navigate to /usr/share/stormond
3. Verify that the file created in the host is seen here
4. Add text to this file
5. Exit the docker container
6. `cat` the file on the host and verify that the text you added is seen here

```
root@str2-7050cx3-acs-13:/host# cd pmon/
root@str2-7050cx3-acs-13:/host/pmon# ls
stormond
root@str2-7050cx3-acs-13:/host/pmon# cd stormond/
root@str2-7050cx3-acs-13:/host/pmon/stormond# touch ashwin.txt
root@str2-7050cx3-acs-13:/host/pmon/stormond# docker exec -it pmon bash
root@str2-7050cx3-acs-13:/# cd /usr/share/stormond/
root@str2-7050cx3-acs-13:/usr/share/stormond# ls
ashwin.txt
root@str2-7050cx3-acs-13:/usr/share/stormond# ls -al
total 8
drwxr-xr-x 2 root root 4096 Feb 19 04:06 .
drwxr-xr-x 3 root root 4096 Feb 19 02:04 ..
-rw-r--r-- 1 root root    0 Feb 19 04:06 ashwin.txt
root@str2-7050cx3-acs-13:/usr/share/stormond# echo "Bind mount successful?" > ashwin.txt
root@str2-7050cx3-acs-13:/usr/share/stormond# exit
exit
root@str2-7050cx3-acs-13:/host/pmon/stormond# cat ashwin.txt
Bind mount successful?
root@str2-7050cx3-acs-13:/host/pmon/stormond#
```

As a third test:

Run `ls -alirth` in both the `/host/pmon/stormond` dir on the host and `/usr/share/stormond` dir in the pmon container and verify that the inode numbers match.

```
root@str2-7050cx3-acs-13:/host/pmon/stormond# ls -alirth
total 12K
129546 drwxr-xr-x 3 root root 4.0K Feb 19 04:06 ..
129554 drwxr-xr-x 2 root root 4.0K Feb 19 04:06 .
129555 -rw-r--r-- 1 root root   23 Feb 19 04:07 ashwin.txt

root@str2-7050cx3-acs-13:/host/pmon/stormond# docker exec -it pmon bash
root@str2-7050cx3-acs-13:/# cd /usr/share/stormond/
root@str2-7050cx3-acs-13:/usr/share/stormond# ls -alirth
total 12K
 33618 drwxr-xr-x 3 root root 4.0K Feb 19 02:04 ..
129554 drwxr-xr-x 2 root root 4.0K Feb 19 04:06 .
129555 -rw-r--r-- 1 root root   23 Feb 19 04:07 ashwin.txt
root@str2-7050cx3-acs-13:/usr/share/stormond#

```

#### Which release branch to backport (provide reason below if selected)

- [ ] 201811
- [ ] 201911
- [ ] 202006
- [ ] 202012
- [ ] 202106
- [ ] 202111
- [ ] 202205
- [ ] 202211
- [ ] 202305

#### Tested branch (Please provide the tested image version)

- [x] custom master image

#### Description for the changelog
adds bind mount from `/host/pmon/stormond` on the host to `/usr/share/stormond` on the pmon container as part of the larger stormon daemon feature.

### CODE REVIEW

File: dockers/docker-platform-monitor/Dockerfile.j2 [hunk 1]
```diff
@@ -91,6 +91,7 @@ COPY ["files/supervisor-proc-exit-listener", "/usr/bin"]
 COPY ["critical_processes", "/etc/supervisor"]
 COPY ["etc/rsyslog.conf", "/etc/rsyslog.conf"]
 
+RUN mkdir -p /host/storagemon
```
Comment (saiarcot895) [inline dockers/docker-platform-monitor/Dockerfile.j2:94]: Having a `/host` directory inside the container doesn't make much sense to me, can you pick a different directory? `/usr/share/storagemon`, perhaps?

Comment (assrinivasan) [reply dockers/docker-platform-monitor/Dockerfile.j2:94]: Oh wait did you mean bind /host/pmon/storagemon on the host to /usr/share/storagemon in the container? If so, yes, agreed -- will make that change.

Comment (saiarcot895) [reply dockers/docker-platform-monitor/Dockerfile.j2:94]: Yes, I meant within the container.

Comment (assrinivasan) [reply dockers/docker-platform-monitor/Dockerfile.j2]: Done in latest commit. @saiarcot895

File: rules/docker-platform-monitor.mk [hunk 2]
```diff
@@ -51,6 +51,7 @@ $(DOCKER_PLATFORM_MONITOR)_CONTAINER_NAME = pmon
 $(DOCKER_PLATFORM_MONITOR)_RUN_OPT += --privileged -t
 $(DOCKER_PLATFORM_MONITOR)_RUN_OPT += -v /etc/sonic:/etc/sonic:ro
 $(DOCKER_PLATFORM_MONITOR)_RUN_OPT += -v /etc/timezone:/etc/timezone:ro 
+$(DOCKER_PLATFORM_MONITOR)_RUN_OPT += -v /host/pmon/storagemon:/host/storagemon:rw
```
Comment (saiarcot895) [inline rules/docker-platform-monitor.mk:54]: This line still needs to be changed.

Comment (assrinivasan) [reply rules/docker-platform-monitor.mk]: Thanks! Done.

**Review by saiarcot895 — APPROVED** [review]

File: dockers/docker-platform-monitor/Dockerfile.j2 [hunk 3]
```diff
@@ -48,6 +48,9 @@ RUN pip3 install requests
 # We install the libpci module in order to be able to do PCI transactions
 RUN pip3 install libpci
 
+# Install psutil for process and system monitoring operations
+RUN pip3 install psutil==5.9.8
```
Comment (saiarcot895) [inline dockers/docker-platform-monitor/Dockerfile.j2:52]: I don't think it needs to be pinned to 5.9.8?

Comment (assrinivasan) [reply dockers/docker-platform-monitor/Dockerfile.j2]: Fixed in latest

**Review by saiarcot895 — APPROVED** [review]

**Comment by prgeor** [comment]
@assrinivasan please point to the HLD in the PR descripton

**Comment by assrinivasan** [comment]
> @assrinivasan please point to the HLD in the PR descripton

Done!

**Review by prgeor — DISMISSED** [review]

**Comment by assrinivasan** [comment]
@saiarcot895 @prgeor I'm not sure why we are seeing an ms_conflict issue. I have verified that the three files I am changing here do not have any conflicts with the corresponding internal repo files.

**Comment by assrinivasan** [comment]
@lguohan all checks have now passed. Could you please help merge? 
cc @prgeor

**Pull request merged**